parteh/PRTAllometricCarbonMod.F90

```diff
@@ -0,0 +1,981 @@
+             write(fates_log(),*) 'exiting'
+             write(fates_log(),*) 'cbal: ',carbon_balance
+             write(fates_log(),*) 'near-zero',nearzero
+             write(fates_log(),*) 'store_c: ',store_c
+             write(fates_log(),*) 'target c: ',target_store_c
+             write(fates_log(),*) 'store_c0:', store_c0
+             call endrun(msg=errMsg(sourcefile, __LINE__))
+          end if
+
+
+          call TargetAllometryCheck(leaf_c, fnrt_c, sapw_c, &
+                                    store_c, struct_c,       &
+                                    target_leaf_c, target_fnrt_c, &
+                                    target_sapw_c, target_store_c, target_struct_c, &
+                                    grow_leaf, grow_fnrt, grow_sapw, grow_store)
+       else
+          grow_leaf  = .true.
+          grow_fnrt = .true.
+          grow_sapw  = .true.
+          grow_store = .true.
+       end if
+
+       ! Initialize the adaptive integrator arrays and flags
+       ! -----------------------------------------------------------------------------------
+       ierr             = 1
+       totalC           = carbon_balance
+       nsteps           = 0
+       
+       c_pool(:) = 0.0_r8
+       c_mask(:) = .false.
+
+       c_pool(leaf_c_id)   = leaf_c
+       c_pool(fnrt_c_id)   = fnrt_c
+       c_pool(sapw_c_id)   = sapw_c
+       c_pool(store_c_id)  = store_c
+       c_pool(struct_c_id) = struct_c
+       c_pool(repro_c_id)  = repro_c
+       c_pool(dbh_id)      = dbh
+       
+       c_mask(leaf_c_id)   = grow_leaf
+       c_mask(fnrt_c_id)   = grow_fnrt
+       c_mask(sapw_c_id)   = grow_sapw
+       c_mask(store_c_id)  = grow_store
+       c_mask(struct_c_id) = .true.                ! Always increment dead on growth step
+       c_mask(repro_c_id)  = .true.                ! Always calculate reproduction on growth
+       c_mask(dbh_id)      = .true.                ! Always increment dbh on growth step
+       
+       if(ODESolve == 2) then
+          this%ode_opt_step = totalC
+       end if
+       
+       do while( ierr .ne. 0 )
+          
+          deltaC = min(totalC,this%ode_opt_step)
+          if(ODESolve == 1) then
+             call RKF45(AllomCGrowthDeriv,c_pool,c_mask,deltaC,totalC, &
+                   max_trunc_error,intgr_params,c_pool_out,this%ode_opt_step,step_pass)
+             
+          elseif(ODESolve == 2) then
+             call Euler(AllomCGrowthDeriv,c_pool,c_mask,deltaC,totalC,intgr_params,c_pool_out)
+             !  step_pass = .true.
+             call CheckIntegratedAllometries(c_pool_out(dbh_id),ipft,canopy_trim,  &
+                   c_pool_out(leaf_c_id), c_pool_out(fnrt_c_id), c_pool_out(sapw_c_id), &
+                   c_pool_out(store_c_id), c_pool_out(struct_c_id), &
+                   c_mask(leaf_c_id), c_mask(fnrt_c_id), c_mask(sapw_c_id), &
+                   c_mask(store_c_id),c_mask(struct_c_id),  max_trunc_error, step_pass)
+             if(step_pass)  then
```

Comment:
What is this bit doing? Is it reducing the tolerance as the number of iterations increases?

Reply:
yup.  I added more comments to explain what is going on.